dbt/adapters/impala/connections.py

```diff
@@ -159,27 +160,51 @@ def open(cls, connection):
                     port=credentials.port,
                     retries=credentials.retries,
                 )
+            connection_end_time = time.time()
 
-            connection.state = "open"
+            connection.state = ConnectionState.OPEN
             connection.handle = handle
         except Exception as ex:
             LOGGER.debug("Connection error {}".format(ex))
-            connection.state = "fail"
+            connection.state = ConnectionState.FAIL
             connection.handle = None
+            connection_end_time = time.time()
             pass
 
         # track usage
         payload = {
             "id": "dbt_impala_open",
-            "unique_hash": hashlib.md5(credentials.host.encode()).hexdigest(),
+            "unique_hash": hashlib.md5((str(credentials.host) + str(credentials.username)).encode()).hexdigest(),
             "auth": auth_type,
+            "connection_name": connection.name,
             "connection_state": connection.state,
+            "connection_opening_time": connection_end_time - connection_start_time,
         }
 
         tracker.track_usage(payload)
 
         return connection
 
+    @classmethod
+    def close(cls, connection):
+        try:
+            connection_start_time = time.time()
```

Comment:
Please rename as connection_close_start_time or something like that. Current name is misleading in the close function.

Reply:
done.